src/main/java/edu/ucdavis/dss/ipa/api/components/budget/BudgetViewController.java

```diff
@@ -230,6 +230,28 @@ public Budget updateBudget(@PathVariable long budgetId,
         return budgetService.update(budgetDTO);
     }
 
+    @RequestMapping(value = "/api/budgetView/budgetScenarios/{budgetScenarioId}", method = RequestMethod.PUT, produces="application/json")
+    @ResponseBody
+    public BudgetScenario updateBudgetScenario(@PathVariable long budgetScenarioId,
+                               @RequestBody BudgetScenario newBudgetScenario,
+                               HttpServletResponse httpResponse) {
+        // Ensure valid params
+        BudgetScenario budgetScenario = budgetScenarioService.findById(budgetScenarioId);
+
+        if (budgetScenario == null) {
+            httpResponse.setStatus(HttpStatus.NOT_FOUND.value());
+            return null;
+        }
+
+        // Authorization check
+        Long workGroupId = budgetScenario.getBudget().getSchedule().getWorkgroup().getId();
```

Comment:
The amount of unchecked, possibly null references makes me uncomfortable UNLESS each of these is annotated @NotNull. Can we ensure that's true?

Reply:
For this particular reference chain, they are all annotated as @NotNull.